thrust/examples/sum_columns.cu

```diff
@@ -0,0 +1,77 @@
+#include <thrust/equal.h>
+#include <thrust/execution_policy.h>
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/discard_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+#include <thrust/random.h>
+#include <thrust/reduce.h>
+#include <thrust/transform.h>
+#include <thrust/universal_vector.h>
+
+#include <cuda/std/mdspan>
+
+#include <iomanip>
+#include <iostream>
+
+int main()
+{
+  const int rows = 32;
+  const int cols = 16;
+
+  // Create a 2D multidimensional array of ints.
+  thrust::universal_vector<int> data(rows * cols, 42);
+  cuda::std::mdspan M(thrust::raw_pointer_cast(data.data()), rows, cols);
+
+  // Create a range to the linear index space for the multidimensional array.
+  auto idxs_begin = thrust::make_counting_iterator(0);
+  auto idxs_end   = idxs_begin + M.size();
+
+  // Fill the array with pseudorandom inputs in parallel on the device.
+  thrust::transform(thrust::device, idxs_begin, idxs_end, M.data_handle(), [] __host__ __device__(int ij) {
+    thrust::default_random_engine rng;
+    thrust::uniform_int_distribution<int> dist(0, 3);
```

Comment:
Should you wait for #4410 to get merged? Are we not going to be deprecating `thrust::uniform_int_distribution`?

Reply:
Thrust's random facility has some extensions for parallelism that I'm not sure that the standard version has. If and when we do deprecated `thrust::uniform_int_distribution` we can revisit replacing it in all of the Thrust examples, not just this one.